src/Frontend/Core.hs

```diff
@@ -221,52 +235,113 @@ percentVotes idea numVoters vv = {- assert c -} v
           else (numVotes idea vv * 100) `div` numVoters
 
 
--- * render context
-
--- | Contains all the information which is needed to render a user role dependent functionality.
-data RenderContext = RenderContext
-      { _renderContextUser     :: User
-      }
-  deriving (Eq, Read, Show)
-
--- | Calculates the render context for role sensitive page rendering
-renderContext :: (ActionPersist m, ActionUserHandler m) => m RenderContext
-renderContext = RenderContext <$> currentUser
-
-
 -- * pages
 
+data AccessResult
```

Comment:
can you shuffle the definitions in this section to tell a story and match the export list?